### PR TITLE
GET and POST server functionality for Categories Table

### PR DESCRIPTION
**Pull request by alambrow**
#### Implemented Get_All_Categories and Create_Categories features

1. Added file `categories.py` to `categories` directory.
2. Modified file `request_handler.py` to include `get_all_categories` and `create_categories` functionality.

#### Steps to Review

1. Checkout this branch locally.
   ```
   git fetch --all
   git checkout al-category-branch
   ```
2. Ensure the backend is  running.
4. Test app functionality.
   > On Postman, run GET and POST fetch calls:
   > GET: http://localhost:8088/categories
   > POST: http://localhost:8088/categories (e.g. {"label": "very interesting"})
5. View code file.
   > Confirm file modifications are present as indicated above.
   > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

**Review by mecrowley — APPROVED** [review]
Passes all tests. File modifications are present as indicated above. No unused code or extraneous comments exist.

**Comment by enbeec** [comment]
Tests run just fine! Moving on to reading.

File: rare-init.sql [hunk 1]
```diff
@@ -85,9 +85,11 @@ CREATE TABLE "Categories" (
...
+INSERT INTO Categories ("id", "label")
+VALUES (1, "News");
 INSERT INTO Tags ('label')
 VALUES ('JavaScript');
 INSERT INTO Reactions ('label', 'image_url')
-VALUES ('happy', 'https://pngtree.com/so/happy');
\ No newline at end of file
+VALUES ('happy', 'https://pngtree.com/so/happy');
+
+SELECT * FROM Categories;
```
Comment (enbeec) [inline rare-init.sql]: This seems extraneous -- could we move this to a separate .sql file like @mecrowley suggested?

File: request_handler.py [hunk 2]
```diff
@@ -72,12 +74,41 @@ def do_GET(self):
...
+        post_body = json.loads(post_body)
+
+        # Parse the URL
+        (resource, id) = self.parse_url(self.path)
+
+        new_category = None
+
+        if resource == "categories":
+            new_category = create_category(post_body)
+            self.wfile.write(json.dumps(new_category).encode())
```
Comment (enbeec) [inline request_handler.py:95]: Awesome. I think we should change do_POST for a user to use the same structure.

**Review by enbeec — APPROVED** [review]
Yay! Looks awesome.

**Pull request merged**